spinalcordtoolbox/deepseg/models.py

```diff
@@ -237,7 +283,30 @@
                              'and Dimitry Van De Ville of EPFL, with the files consisting of lumbar T2w scans (and '
                              'manual spinal cord segmentations) of 11 healthy (non-pathological) patients.',
          'url': 'https://github.com/ivadomed/lumbar_seg_EPFL',
-         'models': ['model_seg_epfl_t2w_lumbar_sc']}
+         'models': ['model_seg_epfl_t2w_lumbar_sc']},
+    'seg_sc_contrast_agnostic':
+        {'description': 'Spinal cord segmentation that is agnostic to contrast',
+         'long_description': 'This segmentation model for contrast agnostic spinal cord segmentation uses an NNUnet '
+                             'architecture, and was created with the MONAI package. Training data was taken from the '
+                             'Spine Generic Multi Subject dataset, with the 6 different contrasts used spanning 267 '
+                             'different healthy (non-pathological) patients.',
+         'url': 'https://github.com/sct-pipeline/contrast-agnostic-softseg-spinalcord/',
+         'models': ['model_seg_sc_contrast_agnostic_softseg_nnunet']},
+    'seg_sc_lesion_t2w_sci':
+        {'description': 'Traumatic SCI spinal cord/lesion seg for T2w contrast',
+         'long_description': 'TBD.',
+         'url': 'https://github.com/ivadomed/model_seg_sci',
+         'models': ['model_seg_sci_multiclass_sc_lesion_nnunet']},
+    'seg_spinal_rootlets_t2w':
+        {'description': 'Segmentation of spinal nerve rootlets for T2w contrast',
+         'long_description': 'TBD.',
+         'url': 'https://github.com/ivadomed/model-spinal-rootlets',
+         'models': ['model_seg_spinal_rootlets_nnunet']},
+    'seg_mouse_gm_wm_t1w':
+        {'description': 'Exvivo mouse GM/WM segmentation for T1w contrast',
+         'long_description': 'TBD.',
+         'url': 'https://github.com/ivadomed/model_seg_mouse-sc_wm-gm_t1',
+         'models': ['model_seg_gm_wm_mouse_nnunet']},
```

Comment:
Good news! I was able to add the Zurich Mouse model with only some minor changes needed (see: https://github.com/spinalcordtoolbox/spinalcordtoolbox/pull/4345/commits/fc0bc4a968156b0769d005b6a3f21112c508a82f, https://github.com/spinalcordtoolbox/spinalcordtoolbox/pull/4345/commits/7ae5838249317bb18df579061418c78129257a4d). I wasn't able to test the inference, though, given the long runtimes: . But, the model loading works, as the script structure was very similar to the existing rootlets/SCI models.

@plbenveniste Would you be able to test out this branch on your private mouse data if possible?

- `git clone` the SCT repo
- `cd spinalcordtoolbox && git checkout jn/4182-integrate-nnunet-monai-into-sct_deepseg`
- Once you're on this branch, run `./install_sct -iy`
- After the installer finishes, run `sct_deepseg -install-task seg_mouse_gm_wm_t1w`
- Then, you should be able to run `sct_deepseg -task seg_mouse_gm_wm_t1w -i {IMG_NAME_HERE}`

Given that there are two labels (GM: 1, WM: 2), I followed the convention of the existing SCI lesion/SC multiclass model by splitting the image into _GM_seg and _WM_seg based on the label values (https://github.com/spinalcordtoolbox/spinalcordtoolbox/commit/7ae5838249317bb18df579061418c78129257a4d). This is slightly different behavior than the original `test.py` script, which I think only outputs 1 image, so that's something we might want to discuss?

We could also follow the convention of the SCI `dataset.json` file by listing "`sc: [1, 2]`" in the list of labels, which will automatically combine the WM/GM labels into a third `_sc_seg` output image, too.

Reply:
@plbenveniste provided me with a sample mouse image via Slack, and I wrote a test for this image here: https://github.com/spinalcordtoolbox/spinalcordtoolbox/pull/4345/commits/73d677b9f1b81b80bbfe29a23516952a7e4ef930

However, I've written it according to @mguaypaq's suggestion: Don't include the file in `sct_testing_data`, skip the test if the file is not present, then provide a link to it so people can manually test locally.

Some questions:

- Are the file size (~10MB) and processing time (~3m18s) truly prohibitive? (I think so, but I'd love to get opinions on this.)
- Where should we host the image? (Can the image even be uploaded publicly?)

Reply:
See https://github.com/spinalcordtoolbox/spinalcordtoolbox/pull/4345#issuecomment-1942072225:

> So, I think what we should do is, for now, put a link to the Wiki page in the test comment (since we can update the Wiki page without tying our data management processes to the 6.2 release deadline). Then, at our next SCT meeting, we can discuss how we should manage our internal testing datasets moving forward.